api/agent/drivers/docker/docker_client.go

```
@@ -246,117 +249,35 @@ func RegisterViews(tagKeys []string, latencyDist []float64) {
 	}
 }
 
-func (d *dockerWrap) retry(ctx context.Context, f func() error) error {
-	var i uint64
-	var err error
-	defer func() { stats.Record(ctx, dockerRetriesMeasure.M(int64(i))) }()
-
-	logger := common.Logger(ctx)
-	var b common.Backoff
-	// 10 retries w/o change to backoff is ~13s if ops take ~0 time
-	for ; i < d.maxRetries; i++ {
-		select {
-		case <-ctx.Done():
-			stats.Record(ctx, dockerTimeoutMeasure.M(0))
-			logger.WithError(ctx.Err()).Warnf("docker call timed out")
-			return ctx.Err()
-		default:
-		}
-
-		err = filter(ctx, f())
-		if common.IsTemporary(err) || isDocker50x(err) {
-			logger.WithError(err).Warn("docker temporary error, retrying")
-			b.Sleep(ctx)
-			continue
-		}
-		if err != nil {
-			stats.Record(ctx, dockerErrorMeasure.M(0))
-		}
-		return err
-	}
-	return err // TODO could return context.DeadlineExceeded which ~makes sense
-}
-
-func isDocker50x(err error) bool {
-	derr, ok := err.(*docker.Error)
-	return ok && derr.Status >= 500
-}
-
-// implement common.Temporary()
-type temporary struct {
-	error
-}
-
-func (t *temporary) Temporary() bool { return true }
-
-func temp(err error) error {
-	return &temporary{err}
-}
-
-// some 500s are totally cool
-func filter(ctx context.Context, err error) error {
-	log := common.Logger(ctx)
-	// "API error (500): {\"message\":\"service endpoint with name task-57d722ecdecb9e7be16aff17 already exists\"}\n" -> ok since container exists
-	switch {
-	default:
-		return err
-	case err == nil:
-		return err
-	case strings.Contains(err.Error(), "service endpoint with name"):
-	}
-	log.WithError(err).Warn("filtering error")
-	return nil
-}
-
-func filterNoSuchContainer(ctx context.Context, err error) error {
```

Dead code. Caller of WaitContainer ignore any errors anyway.